pkg/gadgets/tcptop/gadget.go

```diff
@@ -0,0 +1,228 @@
+	t := `tcptop shows command generating TCP connections, with container details.
+
+The following parameters are supported:
+- interval: Output interval, in seconds. (default %d)
+- max_rows: Maximum rows to print. (default %d)
+- sort: The field to sort the results by (%s). (default %s)
+- filtered_pid: Only get events for this PID (default to all).
+- filtered_family: Only get events for this IP version. (either 4 or 6, default to all)`
+	return fmt.Sprintf(t, intervalDefault, maxRowsDefault,
+		strings.Join(types.SortBySlice, ","), sortByDefault)
+}
+
+func (f *TraceFactory) OutputModesSupported() map[string]struct{} {
+	return map[string]struct{}{
+		"Stream": {},
+	}
+}
+
+func deleteTrace(name string, t interface{}) {
+	trace := t.(*Trace)
+	if trace.tracer != nil {
+		trace.tracer.Stop()
+	}
+}
+
+func (f *TraceFactory) Operations() map[string]gadgets.TraceOperation {
+	n := func() interface{} {
+		return &Trace{
+			resolver: f.Resolver,
+		}
+	}
+
+	return map[string]gadgets.TraceOperation{
+		"start": {
+			Doc: "Start tcptop gadget",
+			Operation: func(name string, trace *gadgetv1alpha1.Trace) {
+				f.LookupOrCreate(name, n).(*Trace).Start(trace)
+			},
+		},
+		"stop": {
+			Doc: "Stop tcptop gadget",
+			Operation: func(name string, trace *gadgetv1alpha1.Trace) {
+				f.LookupOrCreate(name, n).(*Trace).Stop(trace)
+			},
+		},
+	}
+}
+
+func (t *Trace) Start(trace *gadgetv1alpha1.Trace) {
+	if t.started {
+		trace.Status.State = "Started"
+		return
+	}
+
+	traceName := gadgets.TraceName(trace.ObjectMeta.Namespace, trace.ObjectMeta.Name)
+
+	maxRows := maxRowsDefault
+	intervalSeconds := intervalDefault
+	sortBy := sortByDefault
+	targetPid := int32(-1)
+	targetFamily := int32(-1)
+
+	if trace.Spec.Parameters != nil {
+		params := trace.Spec.Parameters
+		var err error
+
+		if val, ok := params["max_rows"]; ok {
+			maxRows, err = strconv.Atoi(val)
+			if err != nil {
+				trace.Status.OperationError = fmt.Sprintf("%q is not valid for ouput_rows", val)
+				return
+			}
+		}
+
+		if val, ok := params["interval"]; ok {
+			intervalSeconds, err = strconv.Atoi(val)
+			if err != nil {
+				trace.Status.OperationError = fmt.Sprintf("%q is not valid for interval", val)
+				return
+			}
+		}
+
+		if val, ok := params["sortby"]; ok {
+			sortBy, err = types.ParseSortBy(val)
+			if err != nil {
+				trace.Status.OperationError = fmt.Sprintf("%q is not valid for sortby", val)
+				return
+			}
+		}
+
+		if pid, ok := params["pid"]; ok {
+			pidParsed, err := strconv.ParseInt(pid, 10, 32)
+			if err != nil {
+				trace.Status.OperationError = fmt.Sprintf("%q is not valid for PID", pid)
+				return
+			}
+
+			targetPid = int32(pidParsed)
+		}
+
+		if family, ok := params["family"]; ok {
+			familyParsed, err := strconv.ParseInt(family, 10, 32)
+			if err != nil {
+				trace.Status.OperationError = fmt.Sprintf("%q is not valid for PID", family)
+				return
+			}
+
+			targetFamily = int32(familyParsed)
```

Comment:
Here you are not validating this value, this is where I said that having `trasnlateFamily` in types.go could be useful. BTW, it could be `PaseIPFamily` to be aligned with `ParseSortBy`.